### PR TITLE
feature/9733-VAIconWithTextMigration

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Migrated tab bar to the design system icons along with header banner right button and menu view that used VAIconWithText.

Renamed VAIconWithText to IconWithText and removed assets

## Screenshots/Video
Screenshot 2024-10-01 at 10 44 00 AM
Screenshot 2024-10-01 at 10 44 09 AM
Screenshot 2024-10-01 at 10 44 19 AM
Screenshot 2024-10-01 at 10 44 24 AM
Screenshot 2024-10-01 at 10 44 27 AM
Screenshot 2024-10-01 at 10 44 32 AM
Screenshot 2024-10-01 at 10 44 36 AM
Screenshot 2024-10-01 at 10 44 45 AM
Screenshot 2024-10-01 at 10 45 09 AM
Screenshot 2024-10-01 at 10 45 14 AM
Screenshot 2024-10-01 at 10 45 26 AM
Screenshot 2024-10-01 at 10 45 30 AM
Screenshot 2024-10-01 at 10 45 35 AM
Screenshot 2024-10-01 at 10 45 41 AM
Screenshot 2024-10-01 at 10 45 56 AM
Screenshot 2024-10-01 at 10 46 01 AM

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
VAIconWithText and all it's usage is updated and assets removed

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Comment by alexandec** [comment]
A couple conflicts on this one

File: VAMobile/src/components/Menu/MenuView.tsx [hunk 1]
```diff
@@ -29,11 +31,9 @@ export type MenuItemActionsType = {
   /** adds a divider after the action*/
   addDivider: boolean
   /** name of the icon to show */
```
Comment (alexandec) [inline VAMobile/src/components/Menu/MenuView.tsx:33]: Comment isn't updated to match the change

File: VAMobile/src/components/Menu/MenuView.tsx [hunk 2]
```diff
@@ -29,11 +31,9 @@ export type MenuItemActionsType = {
...
-  iconName?: keyof typeof VA_ICON_MAP
+  iconProps?: IconProps
   /** method to  */
```
Comment (alexandec) [inline VAMobile/src/components/Menu/MenuView.tsx:35]: This existing comment must have been truncated previously

**Review by alexandec — DISMISSED** [review]
Noticed a couple comments that can be improved

File: VAMobile/src/components/Menu/MenuView.tsx (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/src/components/Menu/MenuView.tsx:35]: updated

File: VAMobile/src/components/Menu/MenuView.tsx (hunk 1 above)
Comment (Sparowhawk) [reply VAMobile/src/components/Menu/MenuView.tsx]: updated

**Review by alexandec — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Comment by timwright12** [comment]
@Sparowhawk is this still with QA, it looks like @rbontrager approved it

**Comment by Sparowhawk** [comment]
@timwright12 It is just pending a FE engineer approval to merge

**Review by theodur — APPROVED** [review]

**Pull request merged**